### ISSUE TITLE
Export org/role as CSV

### ISSUE THREAD

**Issue by Gaetanbrl** [georchestra/georchestra#2730]
We can export users as CSV/VCF from #2278 by @landryb.
This issue is about org/role export as CSV.

**Comment by Gaetanbrl** [comment]
@landryb is it correct for you (org as csv) ?

![image](https://user-images.githubusercontent.com/16317988/66225802-a0b4b580-e6d9-11e9-84c9-1f85b9571fbd.png)

**Comment by landryb** [comment]
Yup, except it lacks the description field showing the area codes.

**Comment by RemiDesgrange** [comment]
@Gaetanbrl encoding problem ?

**Comment by Gaetanbrl** [comment]
@RemiDesgrange #2737 - same for user

**Comment by Gaetanbrl** [comment]
@landryb what is the maximum number of area codes per user ?

**Comment by landryb** [comment]
no limit, in my case 4100 but an sdi could be on france and have 36k...

**Comment by Gaetanbrl** [comment]
Are you sure to display all area code in the CSV ?

**Comment by landryb** [comment]
that's the most useful information to me. And actually, it might be more useful/usable in calc/office if displayed as one line per combination of values instead of concatenating multivalued attributes, ie 20 lignes for an org that has 1 member and an area of 20 codes, 10 lines for an org with 2 members and an area of 5 codes, etc....
@fvanderbiest what do you think ?

**Comment by fvanderbiest** [comment]
> it might be more useful/usable in calc/office if displayed as one line per combination of values instead of concatenating multivalued attributes, ie 20 lignes for an org that has 1 member and an area of 20 codes, 10 lines for an org with 2 members and an area of 5 codes, etc....

Seems weird...
I'd rather expect something like this (eventually with quotes, etc):
```csv
# id;name;members;featureIdList...
c2c;Camptocamp;Remi,François,Pierre,Antoine,Christophe;73000,38500,73370
...
```
Please note that @RemiDesgrange is currently working on replacing the description field by a featureId field (see #2638)

Also : please consider using commons-csv as we did with #2733

**Comment by landryb** [comment]
> Please note that @RemiDesgrange is currently working on replacing the description field by a featureId field (see #2638)

well it would be nice to know what's the actual plan (ie how it will be laid in ldap, where does the featureId come from, etc etc), because many things rely on it here. And nobody likes spending time on something that will have to be dumped in 3 months :)

> Seems weird...
> I'd rather expect something like this (eventually with quotes, etc):
> 
> ```
> # id;name;members;featureIdList...
> c2c;Camptocamp;Remi,François,Pierre,Antoine,Christophe;73000,38500,73370
> ...
> ```

that's a nightmare to use in a spreadsheet. Unusable wrt filters etc. and cells with 4000 values will be unusable too. 

Our usecase is export an org, or all orgs, and (in calc/excel) quickly find which org covers area 73001, how many orgs are of type bar, or in which org is user foo, or export all roles and quickly find which roles the user has, etc. Much simpler if you split the multivalues in distinct lines (like how they would be actual records in a database.... or multivalued attributes in the LDAP)

**Comment by fvanderbiest** [comment]
> how it will be laid in ldap

Right, we plan to have one `featureId` line per insee code. 
See also https://github.com/georchestra/georchestra.github.io/pull/90/files

**Comment by fvanderbiest** [comment]
> Our usecase is export an org, or all orgs, and (in calc/excel) quickly find which org covers area 73001, how many orgs are of type bar, or in which org is user foo, or export all roles and quickly find which roles the user has, etc. Much simpler if you split the multivalues in distinct lines (like how they would be actual records in a database.... or multivalued attributes in the LDAP)

Seems legit then. OK

**Comment by landryb** [comment]
Note to $self, @Gaetanbrl's work in progress is in https://github.com/georchestra/georchestra/compare/master...jdev-org:issue-2730

**Comment by Gaetanbrl** [comment]
@landryb this work in progress not take into account this PR  #2733 merged 3 days ago...
